File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.py
from collections import deque

def solution(queue1, queue2):
    answer = 0
    q1 = deque(queue1)
    q2 = deque(queue2)
    
    t1 = sum(queue1)
    t2 = sum(queue2)
    
    total = t1 + t2
    if total % 2 == 1:
        return -1
    
    target = total // 2
    for t in q1:
        if t > target:
            return -1
        
    while answer <= 6000000:
        if t1 > t2:
            tmp = q1.popleft()
            q2.append(tmp)
            t1 -= tmp
            t2 += tmp
            answer += 1
        elif t1 < t2:
            tmp = q2.popleft()
            q1.append(tmp)
            t1 += tmp
            t2 -= tmp
            answer += 1
        else:
            return answer
    
    return -1

File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.java
import java.util.Queue;
import java.util.LinkedList;

class Solution {
    public int solution(int[] queue1, int[] queue2) {
        int answer = 0;
        Queue<Integer> q1 = new LinkedList<>();
        Queue<Integer> q2 = new LinkedList<>();
        long t1 = 0;
        long t2 = 0;
        int max = 0;
        for (int i = 0; i < queue1.length; i++) {
            q1.add(queue1[i]);
            max = Math.max(max, queue1[i]);
            t1 += queue1[i];
            q2.add(queue2[i]);
            t2 += queue2[i];
        }
        
        long total = t1 + t2;
        if (total % 2 == 1) {
            return -1;
        }
        if (max > total / 2) {
            return -1;
        }
        
        for (int cnt = 0; cnt <= 600000; cnt++) {
            if (t1 > t2) {
                int tmp = q1.poll();
                q2.add(tmp);
                t1 -= tmp;
                t2 += tmp;
            } else if (t1 < t2) {
                int tmp = q2.poll();
                q1.add(tmp);
                t2 -= tmp;
                t1 += tmp;
            } else {
                return answer;
            }
            answer++;
        }
        
        return -1;
    }
}